---
title: "Mycoplasma pneumoniae M547 Annotation"
author: "Joel Pantuso"
date: "4/26/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(dplyr)
library(knitr)
library(utils)
```

```{r data}
# read in data
M547 <- read.csv("SRR2135852.csv")
```

```{r clean}
# clean up data
M547 <- M547 %>%
  filter(Feature != "gene") %>%
  filter(Feature != "pseudogene") %>%
  filter(Feature != "exon")
```


```{r}
# extract useful variables
M547 <- M547 %>%
  select(Feature, Start, End, Strand, Attributes)
write.csv(M547, "M547 Clean.csv", row.names=FALSE)
```

```{r}
# separate out attributes into individual columns
read.table(text=as.character(M547$Attributes), sep=";", fill=TRUE, as.is=TRUE) %>%
  write.csv("attributes.csv", row.names=FALSE)
```

```{r, eval=FALSE}
# Repeat for contigs 7-20, somehow missing from first run
M547.7.20 <- read.csv("contigs7-20.csv")

M547.7.20 <- M547.7.20 %>%
  filter(Feature != "gene") %>%
  filter(Feature != "pseudogene") %>%
  filter(Feature != "exon")

M547.7.20 <- M547.7.20 %>%
  select(Attributes)

read.table(text=as.character(M547.7.20$Attributes), sep=";", fill=TRUE, as.is=TRUE) %>%
  write.csv("attributes7-20.csv", row.names=FALSE)
```

```{r}
# Repeat for contig 6, somehow also missing :/
M547.6 <- read.csv("contig6.csv")

M547.6 <- M547.6 %>%
  filter(Feature != "gene") %>%
  filter(Feature != "pseudogene") %>%
  filter(Feature != "exon")

M547.6 <- M547.6 %>%
  select(Attributes)

read.table(text=as.character(M547.6$Attributes), sep=";", fill=TRUE, as.is=TRUE) %>%
  write.csv("attributes6.csv", row.names=FALSE)
```

